server/komodo-rest/src/main/java/org/komodo/rest/service/KomodoConnectionService.java

```diff
@@ -835,6 +849,204 @@ public Response validateConnectionName( final @Context HttpHeaders headers,
+
+        SecurityPrincipal principal = checkSecurityContext(headers);
+        if (principal.hasErrorResponse())
+            return principal.getErrorResponse();
+
+        List<MediaType> mediaTypes = headers.getAcceptableMediaTypes();
+        if (! isAcceptable(mediaTypes, MediaType.APPLICATION_JSON_TYPE))
+            return notAcceptableMediaTypesBuilder().build();
+
+        // Error if the connection name is missing
+        if (StringUtils.isBlank( connectionName )) {
+            return createErrorResponseWithForbidden(mediaTypes, RelationalMessages.Error.CONNECTION_SERVICE_MISSING_NAME);
+        }
+
+        UnitOfWork uow = null;
+
+        try {
+            uow = createTransaction( principal, "deployVdbForConnection", false ); //$NON-NLS-1$
+
+            // Get the connection
+            Connection connection = findConnection(uow, connectionName);
+            if (connection == null)
+                return commitNoConnectionFound(uow, mediaTypes, connectionName);
+            String jndiName = connection.getJndiName(uow);
+            String driverName = connection.getDriverName(uow);
+            
+            // Name of VDB to be created, based on connection
+            String vdbName = connectionName + CONNECTION_VDB_SUFFIX;
+            
+            // If VDB with name already exists, delete it
+            Repository repo = this.kengine.getDefaultRepository();
+            final WorkspaceManager mgr = WorkspaceManager.getInstance( repo, uow );
+            String repoPath = repo.komodoWorkspace( uow ).getAbsolutePath();
+            
+            final Vdb existingVdb = findVdb( uow, vdbName );
+
+            if ( existingVdb != null ) {
+                mgr.delete(uow, existingVdb);
+            }
+            
+            // Create new VDB
+            String vdbPath = repoPath + "/" + vdbName;
+            final Vdb vdb = getWorkspaceManager(uow).createVdb( uow, null, vdbName, vdbPath );
+            vdb.setDescription(uow, "Vdb for connection "+connectionName);
+                        
+            // Add model to the VDB
+            Model model = vdb.addModel(uow, connectionName);
+            model.setModelType(uow, Model.Type.PHYSICAL);
+            model.setProperty(uow, "importer.TableTypes", "TABLE,VIEW");
+            model.setProperty(uow, "importer.UseQualifiedName", "true");
+            model.setProperty(uow, "importer.UseCatalogName", "false");
+            model.setProperty(uow, "importer.UseFullSchemaName", "false");
+            
+            // Add model source to the model
+            ModelSource modelSource = model.addSource(uow, connectionName);
+            modelSource.setJndiName(uow, jndiName);
+            modelSource.setTranslatorName(uow, driverName);
+            
+            // Deploy the VDB
+            DeployStatus deployStatus = vdb.deploy(uow);
+            
+            // Await the deployment to end
+            Thread.sleep(DEPLOYMENT_WAIT_TIME);
```

Comment:
This code should be shared with KomodoMetadataService rather than duplicated for the same reason.